src/server.rs

```diff
@@ -118,6 +156,17 @@ async fn static_assets(req: Request<Body>) -> (StatusCode, HeaderMap, Body) {
     let mut file = match fs::read(&path) {
         Ok(file) => file,
         Err(err) => {
+            if *index && reading_index {
```

Comment:
Here we do not know yet if that we cannot read the file is because the file does not exist or of other reason. I suggest only show this page when we cannot find the `index.html`.

Reply:
Hmmm. Good point. Would you want to see this moved down to where the `err.kind()` is getting matched, or restructure the whole function?